File: backend/src/main/java/com/lerestaurant/le_restaurant_backend/entity/Order.java
package com.lerestaurant.le_restaurant_backend.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
public class Order {

    public enum OrderType { DINE_IN, TAKEOUT, DELIVERY }
    public enum OrderStatus { PENDING, CONFIRMED, PREPARING, READY, COMPLETED, CANCELLED }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private User customer;

    @ManyToOne
    @JoinColumn(name = "table_id")
    private RestaurantTable table; // nullable for delivery

    @Enumerated(EnumType.STRING)
    @Column(name = "order_type", nullable = false)
    private OrderType orderType = OrderType.DINE_IN;

    @Column(precision = 10, scale = 2)
    private BigDecimal subtotal = BigDecimal.ZERO;

    @Column(name = "tax_amount", precision = 10, scale = 2)
    private BigDecimal taxAmount = BigDecimal.ZERO;

    @Column(name = "tip_amount", precision = 10, scale = 2)
    private BigDecimal tipAmount = BigDecimal.ZERO;

    @Column(name = "total_amount", precision = 10, scale = 2)
    private BigDecimal totalAmount = BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status = OrderStatus.PENDING;

    @Column(name = "special_instructions", columnDefinition = "TEXT")
    private String specialInstructions;

    @Column(name = "order_time", nullable = false)
    private OffsetDateTime orderTime = OffsetDateTime.now();

    @Column(name = "estimated_completion")
    private OffsetDateTime estimatedCompletion;

    @Column(name = "completed_at")
    private OffsetDateTime completedAt;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> items = new ArrayList<>();

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public User getCustomer() { return customer; }
    public void setCustomer(User customer) { this.customer = customer; }
    public RestaurantTable getTable() { return table; }
    public void setTable(RestaurantTable table) { this.table = table; }
    public OrderType getOrderType() { return orderType; }
    public void setOrderType(OrderType orderType) { this.orderType = orderType; }
    public BigDecimal getSubtotal() { return subtotal; }
    public void setSubtotal(BigDecimal subtotal) { this.subtotal = subtotal; }
    public BigDecimal getTaxAmount() { return taxAmount; }
    public void setTaxAmount(BigDecimal taxAmount) { this.taxAmount = taxAmount; }
    public BigDecimal getTipAmount() { return tipAmount; }
    public void setTipAmount(BigDecimal tipAmount) { this.tipAmount = tipAmount; }
    public BigDecimal getTotalAmount() { return totalAmount; }
    public void setTotalAmount(BigDecimal totalAmount) { this.totalAmount = totalAmount; }
    public OrderStatus getStatus() { return status; }
    public void setStatus(OrderStatus status) { this.status = status; }
    public String getSpecialInstructions() { return specialInstructions; }
    public void setSpecialInstructions(String specialInstructions) { this.specialInstructions = specialInstructions; }
    public OffsetDateTime getOrderTime() { return orderTime; }
    public void setOrderTime(OffsetDateTime orderTime) { this.orderTime = orderTime; }
    public OffsetDateTime getEstimatedCompletion() { return estimatedCompletion; }
    public void setEstimatedCompletion(OffsetDateTime estimatedCompletion) { this.estimatedCompletion = estimatedCompletion; }
    public OffsetDateTime getCompletedAt() { return completedAt; }
    public void setCompletedAt(OffsetDateTime completedAt) { this.completedAt = completedAt; }
    public List<OrderItem> getItems() { return items; }
    public void setItems(List<OrderItem> items) { this.items = items; }
}

File: frontend/src/types/order.ts
/**
 * Order Types (F105)
 * Data models for order management operations
 * 
 * ⚠️ IMPORTANT: OrderStatus values MUST match backend enum exactly
 * Backend: PENDING, CONFIRMED, PREPARING, READY, COMPLETED, CANCELLED
 */

export type OrderType = 'DINE_IN' | 'TAKEOUT' | 'DELIVERY';
export type OrderStatus = 'PENDING' | 'CONFIRMED' | 'PREPARING' | 'READY' | 'COMPLETED' | 'CANCELLED';

export interface OrderItemDto {
  id: number;
  orderId: number;
  menuItemId: number;
  menuItemName: string;
  unitPrice: number;
  quantity: number;
  subtotal: number;
}

export interface OrderDto {
  id: number;
  customerId: number;
  customerName: string;
  customerEmail: string;
  customerPhone: string;
  tableId?: number;
  tableNumber?: string;
  orderType: OrderType;
  status: OrderStatus;
  items: OrderItemDto[];
  subtotal: number;
  taxAmount: number;
  tipAmount: number;
  totalAmount: number;
  specialInstructions?: string;
  orderTime: string; // ISO datetime
  estimatedCompletion?: string;
  completedAt?: string;
  createdAt: string;
}

export interface OrderItemRequestDto {
  menuItemId: number;
  quantity: number;
}

export interface OrderCreateRequestDto {
  customerId: number;
  tableId?: number;
  orderType: OrderType;
  items: OrderItemRequestDto[];
  specialInstructions?: string;
  tipAmount?: number;
}

export interface OrderUpdateRequestDto {
  status?: OrderStatus;
  specialInstructions?: string;
  tipAmount?: number;
}

export interface OrderFilterOptions {
  customerId?: number;
  status?: OrderStatus;
  orderType?: OrderType;
}
